tests/baselines/reference/logicalAssignment5(target=es2015).errors.txt

```diff
@@ -0,0 +1,51 @@
+tests/cases/conformance/esnext/logicalAssignment/logicalAssignment5.ts(12,12): error TS7006: Parameter 'a' implicitly has an 'any' type.
+tests/cases/conformance/esnext/logicalAssignment/logicalAssignment5.ts(13,5): error TS2722: Cannot invoke an object which is possibly 'undefined'.
+tests/cases/conformance/esnext/logicalAssignment/logicalAssignment5.ts(17,12): error TS2532: Object is possibly 'undefined'.
+tests/cases/conformance/esnext/logicalAssignment/logicalAssignment5.ts(22,12): error TS2532: Object is possibly 'undefined'.
+tests/cases/conformance/esnext/logicalAssignment/logicalAssignment5.ts(27,27): error TS7006: Parameter 'a' implicitly has an 'any' type.
+tests/cases/conformance/esnext/logicalAssignment/logicalAssignment5.ts(28,5): error TS2722: Cannot invoke an object which is possibly 'undefined'.
+
+
+==== tests/cases/conformance/esnext/logicalAssignment/logicalAssignment5.ts (6 errors) ====
+    function foo1 (f?: (a: number) => void) {
+        f ??= (a => a)
+        f(42)
+    }
+    
+    function foo2 (f?: (a: number) => void) {
+        f ||= (a => a)
+        f(42)
+    }
+    
+    function foo3 (f?: (a: number) => void) {
+        f &&= (a => a)
+               ~
+!!! error TS7006: Parameter 'a' implicitly has an 'any' type.

```

Comment:
Shouldn't `a` be contextually typed to `number` here? This expands to `f && (f = (a => a))`, so there's no reason the contextual type shouldn't apply.